product-service/src/handlers/catalogBatchProcess.ts

```diff
@@ -0,0 +1,58 @@
+import { SNS } from 'aws-sdk';
+import { PublishInput } from 'aws-sdk/clients/sns';
+import productProviderDynamoDB from '../dynamodb-providers/product.provider';
+import { HttpResponse } from '../helpers/http-response';
+import { isJson } from '../utils/isJson';
+
+const catalogBatchProcess = (event: any, context: any, cb: any) => {
+    const sns = new SNS();
```

Comment:
Pay attention, that lambda can cache some objects between execution when it's warm.

It's best practice to cache such things as initialized clients in order to reuse it:

`Take advantage of execution environment reuse to improve the performance of your function. Initialize SDK clients and database connections outside of the function handler, and cache static assets locally in the /tmp directory. Subsequent invocations processed by the same instance of your function can reuse these resources. This saves cost by reducing function run time.`
https://docs.aws.amazon.com/lambda/latest/dg/best-practices.html